autoblocks/_impl/tracer.py

```diff
@@ -244,6 +231,45 @@ async def _send_event_unsafe(
         )
         req.raise_for_status()
 
+    def _send_test_event_unsafe(
+        self,
+        # Require all arguments to be specified via key=value
+        *,
+        message: str,
+        timestamp: str,
+        properties: Dict[Any, Any],
+        trace_id: Optional[str],
+    ) -> None:
+        """Sends an event to the Autoblocks CLI in a testing environment."""
+        if not self._cli_server_address:
+            log.error("Failed to send test event to Autoblocks. CLI server address not set.")
+            return
+
+        if not current_external_test_id.get():
+            log.error("Failed to send test event to Autoblocks. No test ID set.")
+            return
+
+        if not current_test_case_hash.get():
+            log.error("Failed to send test event to Autoblocks. No test case hash set.")
+            return
+
+        traced_event = TracerEvent(
+            message=message,
+            trace_id=trace_id,
+            timestamp=timestamp,
+            properties=properties,
+        )
+        req = global_state.sync_http_client().post(
```

Comment:
Would prefer we just use the async client for everything to be consistent but I guess this works if we're doing it on the main thread and not in our event loop

Reply:
Oh wait I guess that wouldn't even work right? It was blocking?

Reply:
Well this function is running in our event loop since it's in the test context. I think we'll always want this to be blocking in the test context

Reply:
Oh well in that case it would be nice to use the async client because when you use a sync client you're blocking the loop until it completes instead of yielding to other tasks

Reply:
Since it's just sending to the local cli server (and that just adds to an in memory array and returns) it should be very fast.

I'd have to use event-loop.execute to use async right? I can see if it can get that work

Reply:
I think in the testing context it doesn't matter too much what we do here. And I think we're going to switch to queue + sync anyway right? So maybe we can leave it

Reply:
Ya I was just opting for simple in test context. I'm not sure if we'll want queue for test events. We'd have to make sure to flush it before we send test case results

Reply:
Ya I meant for prod